api/v1alpha1/ipfscluster_types.go

```diff
@@ -91,6 +105,8 @@ type IpfsClusterSpec struct {
 	// should use when reproviding content.
 	// +optional
 	Reprovider ReprovideSettings `json:"reprovider,omitempty"`
+	Gateway    ExternalSettings  `json:"gateway"`
+	ClusterAPI ExternalSettings  `json:"clusterApi"`
```

Comment:
I think we can make `Gateway` and `ClusterAPI` as optional field.

Reply:
Good call. Setting these to omitempty